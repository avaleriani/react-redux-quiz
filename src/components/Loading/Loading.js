import React from "react";
import styles from "./styles.module.css";

/**
 * @return HTMLElement
 */
const Loading = () => (
  <div className={styles['loading']}>
    <img src={`${process.env.PUBLIC_URL }/images/loading.svg`} alt="loading"/>
  </div>
);

export default Loading;