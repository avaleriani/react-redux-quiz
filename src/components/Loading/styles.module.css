.loading {
    margin: auto;
    width: 250px;
    padding: 60px;
    display: flex;
    align-items: center;
    justify-content: center;
}

.loading img {
    width: 150px;
    height: 150px;
    animation: rotation 2s infinite linear;
}

@keyframes rotation {
    from {
        transform: rotate(0deg);
    }
    to {
        transform: rotate(359deg);
    }
}