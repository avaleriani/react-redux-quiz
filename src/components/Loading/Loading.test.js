import React from 'react';
import renderer from 'react-test-renderer';
import Loading from './Loading';

describe('Loading component', () => {

  test('renders correctly without props', () => {
    const LoadingComponent = renderer.create(<Loading/>).toJSON();
    expect(LoadingComponent).toMatchSnapshot();
  });
});