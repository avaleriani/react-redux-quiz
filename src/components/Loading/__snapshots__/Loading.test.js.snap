// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading component renders correctly without props 1`] = `
<div
  className="loading"
>
  <img
    alt="loading"
    src="/images/loading.svg"
  />
</div>
`;
